vercel/resource_shared_environment_variable_project_link.go

```diff
@@ -0,0 +1,224 @@
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error reading shared environment variable",
+			"Could not read shared environment variable, unexpected error: "+err.Error(),
+		)
+		return
+	}
+
+	result := SharedEnvironmentVariableProjectLink{
+		TeamID:                      types.StringValue(response.TeamID),
+		SharedEnvironmentVariableID: types.StringValue(response.ID),
+		ProjectID:                   state.ProjectID,
+	}
+
+	for _, pid := range response.ProjectIDs {
+		if pid == result.ProjectID.ValueString() {
+			tflog.Info(ctx, "read shared environment variable for linked project", map[string]interface{}{
+				"team_id":                        result.TeamID.ValueString(),
+				"shared_environment_variable_id": result.SharedEnvironmentVariableID.ValueString(),
+				"project_id":                     result.ProjectID.ValueString(),
+			})
+			diags = resp.State.Set(ctx, result)
+			resp.Diagnostics.Append(diags...)
+			return
+		}
+	}
+
+	tflog.Info(ctx, "failed to read shared environment variable for linked project", map[string]interface{}{
+		"team_id":                        result.TeamID.ValueString(),
+		"shared_environment_variable_id": result.SharedEnvironmentVariableID.ValueString(),
+		"project_id":                     result.ProjectID.ValueString(),
+	})
+
+	// not found, so replace state
+	resp.State.RemoveResource(ctx)
+}
+
+func (r *sharedEnvironmentVariableProjectLinkResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	resp.Diagnostics.AddError("Linking should always be recreated", "Something incorrectly caused an Update, this should always be recreated instead of updated.")
+}
+
+func (r *sharedEnvironmentVariableProjectLinkResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var plan SharedEnvironmentVariableProjectLink
+	diags := req.State.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	request, ok := plan.toUpdateSharedEnvironmentVariableRequest(ctx, false)
+	if !ok {
+		return
+	}
+
+	response, err := r.client.UpdateSharedEnvironmentVariable(ctx, request)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error unlinking project from shared environment variable",
+			"Could not unlink project from shared environment variable, unexpected error: "+err.Error(),
+		)
+		return
+	}
+
+	result := SharedEnvironmentVariableProjectLink{
+		TeamID:                      types.StringValue(response.TeamID),
+		SharedEnvironmentVariableID: types.StringValue(response.ID),
+		ProjectID:                   plan.ProjectID,
+	}
+
+	tflog.Info(ctx, "project unlinked from shared environment", map[string]interface{}{
+		"team_id":                        result.TeamID.ValueString(),
+		"shared_environment_variable_id": result.SharedEnvironmentVariableID.ValueString(),
+		"project_id":                     result.ProjectID.ValueString(),
+	})
```

Comment:
Another thought - should this have an Import, so you can terraform import team_xxx/env_yyy/prj_zzz ?

Reply:
hrmmmmmm.. maybe?

I wouldn't really expect to need to import this, and a creation of an existing resource would just be a no-op. I'll be able to test this as I already have projects linked to shared env vars that I would want to put into terraform.